zc_plugins/EmailArchiveManager/v4.0.0/admin/email_archive_manager.php

```diff
@@ -33,11 +40,21 @@ function zen_get_email_archive_search_query(string $search_text, string $sd_raw,
         'email_text',
         'date_sent',
         'module',
-        'errorinfo',
     ];
     $where_clauses = [];
 
-    $archive_search_sql = 'SELECT ' . implode(', ', $select_fields) . ' FROM ' . TABLE_EMAIL_ARCHIVE . " \n";
+    if (!empty($error_keywords)) {
+        $where_clauses[] = "errorinfo LIKE '%$error_keywords%'";
```

Comment:
For consistency of safety, I'd like to see `zen_db_input()` applied to `$error_keywords` here within this function, instead of relying on it to be applied externally before passing "to" this function.

Reply:
Cool, I'm a little erratic with my use of these input sanitizer methods.
I also didn't use `zen_icon` because I'm writing this against my old 1.5.8 live shop, and forgot to bring over some other little 'i' info tooltips for the other search fields. I can add those in now.
As it seems you're looking at it now, from your earlier comment, I can also:
- make errorinfo always visible i.e. remove the 3 radio buttons and turn it back into an 'only errors' checkbox, since if we're always rendering errorinfo it becomes a boolean choice.
- could go back to your approach of searching 'main' fields and 'errorinfo' at the same time.  With a LIKE query, however, it may be more DB demanding as the engine will have ot query that errorinfo column, though if most are NULL maybe it's a low cost operation.
- I think the date pickers ought to default to some short range, for shops with years of data in them, and a dropdown to populate them with 'last 7 days', '30 days', 'this month' etc.  Does jquery-ui offer such a datepicker-related UI component or do we have to roll our own?

Reply:
Yes, the more I look at your 3-way switch I find myself feeling that it's redundant. Let's go back from that.

The LIKE query on main fields and errors is probably semi inconsequential. Not that there won't be minor delays on large databases, but there aren't really any record-locking issues with this table so like you say, I feel like it's low-cost.

Date-picker: yes, this is the jQuery-UI implementation. The JS that fires it is near the bottom of the page.
Do you mind treating the dates idea as a separate PR? Not that it's a big deal, but it does keep the overall PRs more canonical, and allows time to tinker and explore independent of the current changes.